src/compiler/binder.ts

```diff
@@ -2863,7 +2853,7 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
                 return bindPropertyOrMethodOrAccessor(node as Declaration, SymbolFlags.Method | ((node as MethodDeclaration).questionToken ? SymbolFlags.Optional : SymbolFlags.None),

```

Comment:
The fact this still requires a cast indicates the `.questionToken` property access is polymorphic.